src/commands/lint.ts

```diff
@@ -48,37 +56,57 @@ export class LintCommand implements Command {
       )
     },
     {
-      name: 'no-recursion',
+      name: 'follow-dependencies',
+      type: Boolean,
+      description: (
+        'Follow through and lint dependencies. This is default behavior ' +
+        'when linting your entire application via the entrypoint, shell, ' +
+        'and fragment arguments.'
+      )
+    },
+    {
+      name: 'no-follow-dependencies',
       type: Boolean,
       description: (
-        'Only report errors on specified input files, not from their dependencies.'
+        'Only lint the files provided, ignoring dependencies. This is ' +
+        'default behavior when linting a specific list of files provided ' +
+        'via the input argument.'
       )
     }
   ];
 
   run(options, config): Promise<any> {
-    if (config.inputs.length === 0) {
+    let lintFiles: string[] = options.input
```

Comment:
So the only time `options.input` will be an empty list is if you do `polymer lint --input`. It's a strange case but if that did happen I would expect it to lint zero files. Maybe throw an error or log a warning in that case?